In [1]:
import pandas as pd

df = pd.read_csv("superstore_sales_feature_engineered.csv")

df.head()

,Row ID,Order ID,Order Date,Ship Date,Customer ID,Customer Name,Country,City,State,Postal Code,...,Sub-Category_Envelopes,Sub-Category_Fasteners,Sub-Category_Furnishings,Sub-Category_Labels,Sub-Category_Machines,Sub-Category_Paper,Sub-Category_Phones,Sub-Category_Storage,Sub-Category_Supplies,Sub-Category_Tables
0,1,CA-2017-152156,2017-11-08,2017-11-11,CG-12520,Claire Gute,United States,Henderson,Kentucky,42420.0,...,0,0,0,0,0,0,0,0,0,0
1,2,CA-2017-152156,2017-11-08,2017-11-11,CG-12520,Claire Gute,United States,Henderson,Kentucky,42420.0,...,0,0,0,0,0,0,0,0,0,0
2,3,CA-2017-138688,2017-06-12,2017-06-16,DV-13045,Darrin Van Huff,United States,Los Angeles,California,90036.0,...,0,0,0,1,0,0,0,0,0,0
3,4,US-2016-108966,2016-10-11,2016-10-18,SO-20335,Sean O'Donnell,United States,Fort Lauderdale,Florida,33311.0,...,0,0,0,0,0,0,0,0,0,1
4,5,US-2016-108966,2016-10-11,2016-10-18,SO-20335,Sean O'Donnell,United States,Fort Lauderdale,Florida,33311.0,...,0,0,0,0,0,0,0,1,0,0


In [2]:
X = df[
    [
        'Shipping Days',
        'Order Year',
        'Category_Furniture',
        'Category_Office Supplies',
        'Category_Technology'
    ]
]

y = df['Sales']

print("X shape:", X.shape)
print("y shape:", y.shape)

X shape: (9800, 5)
y shape: (9800,)


In [4]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

print("Training data:", X_train.shape)
print("Testing data:", X_test.shape)

Training data: (7840, 5)
Testing data: (1960, 5)


In [5]:
from sklearn.linear_model import LinearRegression

model = LinearRegression()

print("Model created successfully!")

Model created successfully!


In [6]:
model.fit(X_train, y_train)

print("Model trained successfully!")

Model trained successfully!


In [7]:
y_pred = model.predict(X_test)

print("First 10 predictions:")
print(y_pred[:10])

First 10 predictions:
[ 0.17269068 -0.1739849   0.31100525 -0.1902056  -0.17352629 -0.17771023
 -0.1902056  -0.18372861 -0.19112282 -0.1739849 ]


In [8]:
from sklearn.metrics import mean_absolute_error

mae = mean_absolute_error(y_test, y_pred)

print("Mean Absolute Error:", mae)

Mean Absolute Error: 0.4380627453385851


In [9]:
comparison = pd.DataFrame({
    'Actual Sales': y_test.values[:10],
    'Predicted Sales': y_pred[:10]
})

comparison

,Actual Sales,Predicted Sales
0,-0.291770,0.172691
1,-0.350147,-0.173985
2,-0.350801,0.311005
3,-0.340380,-0.190206
4,-0.334571,-0.173526
5,-0.226512,-0.177710
6,-0.357628,-0.190206
7,-0.332401,-0.183729
8,0.033211,-0.191123
9,-0.307225,-0.173985


In [10]:
print("Day 11 ML Model Summary")
print("------------------------")
print("Algorithm: Linear Regression")
print("Training samples:", len(X_train))
print("Testing samples:", len(X_test))
print("Mean Absolute Error:", round(mae, 4))

Day 11 ML Model Summary
------------------------
Algorithm: Linear Regression
Training samples: 7840
Testing samples: 1960
Mean Absolute Error: 0.4381


In [11]:
prediction_output = pd.DataFrame({
    'Actual Sales': y_test.values,
    'Predicted Sales': y_pred
})

prediction_output.to_csv(
    "day11_predictions.csv",
    index=False
)

print("Prediction file saved successfully!")

Prediction file saved successfully!


# Day 11 – Machine Learning Foundations

## Objective
Built a complete baseline Machine Learning workflow using the Superstore dataset.

## ML Workflow
1. Selected input features and target variable
2. Split the dataset into training and testing sets
3. Used Linear Regression as the baseline model
4. Trained the model using 7,840 training samples
5. Generated predictions for 1,960 unseen test samples
6. Evaluated prediction quality using Mean Absolute Error (MAE)

## Features Used
- Shipping Days
- Order Year
- Category_Furniture
- Category_Office Supplies
- Category_Technology

## Target
- Sales

## Model
Linear Regression

## Evaluation
Mean Absolute Error (MAE): 0.4381

## Output
Prediction results were saved as:
`day11_predictions.csv`

## Key Learning
Machine Learning follows a complete workflow:
Data → Train/Test Split → Training → Prediction → Evaluation

A simple baseline model provides a starting point for improving model performance.